src/core/storage/fileio/s3_api.cpp

```diff
@@ -543,7 +566,7 @@ list_objects_response list_objects_impl(s3url parsed_url, std::string proxy,
           logstream(LOG_DEBUG)
               << "list_objects_impl failed:" << ret.error << std::endl;
           // no need to retry, just jump out all while loop
-          break;
+          break;  // or n_retry = 3;
```

Comment:
I don't think this comment is helpful.

Reply:
I think it's good for people to understand the last n_retry = 3

Reply:
It going to be very easy for this comment to get out of date. If someone changes the while-loop conditional but doesn't update this inline comment, this comment will go from confusing to incomprehensible.

Reply:
The scale for `backoff` is ms. This is just a scalar factor.

Reply:
Then they should read my code and change the condition. I use `n_retry==3` in other places. I just want to remind people not to think the previous guy working here only use `break`.

Reply:
I don't understand the comment either.  I don't think people would get that context.